static/src/javascripts/projects/common/modules/video/media-events.js

```diff
@@ -0,0 +1,67 @@
+define([
+    'bean',
+    'common/utils/$',
+    'common/utils/mediator',
+    'lodash/objects/forOwn',
+    'lodash/functions/throttle'
+], function(
+    bean,
+    $,
+    mediator,
+    forOwn,
+    throttle
+) {
+
+    function addContentEvents(el) {
+        var mediaType = el.tagName.toLowerCase();
+        // TODO: fastdom
+        var mediaId = el.getAttribute('data-media-id');
+
+        // This is to ensure the ready event fires, this is only in enhanced mode.
+        // We might want to restrict this more to desktop for data reasons, but we would have to figure what we do about
+        // the analytics.
+        el.preload = 'metadata';
+
+        // Here we are making sure that we only fire these events once so we aren't tracking things a million times.
+        // General UX events e.g. pause, play, fullscreen, will be tacked elsewhere.
+        var fireOnceEventMapping = {
+            canplay: 'ready',
```

Comment:
`canplay` is from the native API - doesn't make that much sense and is fired loads of times. This fires once once we're ready.